src/libpriv/rpmostree-importer.cxx

```diff
@@ -111,21 +111,18 @@ rpmostree_importer_read_metainfo (int fd, rpmostreecxx::RpmImporterFlags &flags,
   g_auto (Header) ret_header = NULL;
   g_auto (rpmfi) ret_fi = NULL;
   gsize ret_cpio_offset;
-  g_autofree char *abspath = g_strdup_printf ("/proc/self/fd/%d", fd);
 
   ts = rpmtsCreate ();
   rpmtsSetVSFlags (ts, _RPMVSF_NOSIGNATURES);
 
-  /* librpm needs Fopenfd */
-  rpmfd = Fopen (abspath, "r.fdio");
+  rpmfd = fdDup (fd);
```

Comment:
Tangentially...librpm really lacks symbol namespacing.  I guess it's just Really Old C code...